src/shared/vlan-util.c

```diff
@@ -61,6 +62,10 @@ int config_parse_vlanid(
                 return 0;
         }
         if (r < 0) {
+                if (streq(lvalue, "DefaultPVID") && streq(rvalue, "none")) {
```

Comment:
uh oh, i really don#t like this special-casing here...

Please introduce a new function config_parse_default_vlanid(), that is primarily a wrapper around config_parse_vlanid(), but before handing things off will check for "none" and handle that directly. Then, use that new function in the netdev-gperf.gperf file for DefaultPVID=, but leave the original function for [VLAN]'s ID=, and things should be much prettier.